src/Xenial.Tasty.Cli/TastyServer.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xenial.Delicious.Metadata;
+using Xenial.Delicious.Protocols;
+using Xenial.Delicious.Reporters;
+using Xenial.Delicious.Utils;
+
+using static Xenial.Delicious.Utils.Actions;
+
+namespace Xenial.Delicious.Cli
+{
+    public class TastyServer
+    {
+        public event EventHandler<ExecuteCommandEventArgs>? ExecuteCommand;
```

Comment:
Do you need multicast here? Doesn't Action<T> suffice?

Reply:
JsonRPC convention 🤔 I didn't try out normal delegates, but there isn't a lot of docs out there.

Reply:
Okay no way to use normal `Actions`. `StreamJsonRpc` only allows events with `EventHandler` and `EventHandler<T>`. So I guess this is fine here (although I think we should abstract away `StreamJsonRpc` in the near future to support also GRPc and other transport channels.

Reply:
And in combination with the null coalesce operator, I think it's fine to declare them as nullable.